service/kvmconfig/framework.go

```diff
@@ -89,118 +84,55 @@ func NewFramework(config FrameworkConfig) (*framework.Framework, error) {
+	var newInformer *informer.Informer
+	{
+		c := informer.DefaultConfig()
 
-		if versionBundleVersion == "1.0.0" {
-			return true
-		}
-		if versionBundleVersion == "0.1.0" {
-			return true
-		}
-		if versionBundleVersion == "" {
-			return true
-		}
+		c.Watcher = config.G8sClient.ProviderV1alpha1().KVMConfigs("")
 
-		return false
+		newInformer, err = informer.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
 	}
 
-	var v2Resources []framework.Resource
+	var v2ResourceSet *framework.ResourceSet
 	{
-		c := v2.ResourcesConfig{
+		c := v2.ResourceSetConfig{
 			CertsSearcher:      certsSearcher,
 			K8sClient:          config.K8sClient,
 			Logger:             config.Logger,
 			RandomkeysSearcher: randomkeysSearcher,
 
+			HandledVersionBundles: []string{
+				"1.1.0",
+			},
 			Name: config.Name,
 		}
 
-		v2Resources, err = v2.NewResources(c)
+		v2ResourceSet, err = v2.NewResourceSet(c)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 	}
 
-	v3ResourceSetHandles := func(obj interface{}) bool {
-		customObject, err := v3key.ToCustomObject(obj)
-		if err != nil {
-			return false
-		}
-		versionBundleVersion := v3key.VersionBundleVersion(customObject)
-
-		if versionBundleVersion == "1.1.0" {
-			return true
-		}
-
-		return false
-	}
-
-	v3InitCtx := func(ctx context.Context, obj interface{}) (context.Context, error) {
-		if config.GuestUpdateEnabled {
-			updateallowedcontext.SetUpdateAllowed(ctx)
-		}
-
-		return ctx, nil
-	}
-
-	var v3Resources []framework.Resource
+	var v3ResourceSet *framework.ResourceSet
 	{
-		c := v3.ResourcesConfig{
+		c := v3.ResourceSetConfig{
 			CertsSearcher:      certsSearcher,
 			K8sClient:          config.K8sClient,
 			Logger:             config.Logger,
 			RandomkeysSearcher: randomkeysSearcher,
 
+			GuestUpdateEnabled: config.GuestUpdateEnabled,
+			HandledVersionBundles: []string{
+				"1.0.0",
+				"0.1.0",
+			},
 			Name: config.Name,
 		}
 
-		v3Resources, err = v3.NewResources(c)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-	}
-
-	var newInformer *informer.Informer
-	{
-		c := informer.DefaultConfig()
-
-		c.Watcher = config.G8sClient.ProviderV1alpha1().KVMConfigs("")
-
-		newInformer, err = informer.New(c)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-	}
-
-	var v2ResourceSet *framework.ResourceSet
-	{
-		c := framework.ResourceSetConfig{}
-
-		c.Handles = v2ResourceSetHandles
-		c.Logger = config.Logger
-		c.Resources = v2Resources
-
-		v2ResourceSet, err = framework.NewResourceSet(c)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-	}
-
-	var v3ResourceSet *framework.ResourceSet
-	{
-		c := framework.ResourceSetConfig{}
-
-		c.Handles = v3ResourceSetHandles
-		c.InitCtx = v3InitCtx
-		c.Logger = config.Logger
-		c.Resources = v3Resources
-
-		v3ResourceSet, err = framework.NewResourceSet(c)
+		v3ResourceSet, err = v3.NewResourceSet(c)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
```

Comment:
The diff is ugly. This is how it really looks like. Same for `v2`. Nothing more to add to framework.go.

```go
	var v3ResourceSet *framework.ResourceSet
	{
		c := v3.ResourceSetConfig{
			CertsSearcher:      certsSearcher,
			K8sClient:          config.K8sClient,
			Logger:             config.Logger,
			RandomkeysSearcher: randomkeysSearcher,

			GuestUpdateEnabled: config.GuestUpdateEnabled,
			HandledVersionBundles: []string{
				"1.1.0",
			},
			Name: config.Name,
		}

		v3ResourceSet, err = v3.NewResourceSet(c)
		if err != nil {
			return nil, microerror.Mask(err)
		}
	}
```